tsconfig.json

```diff
@@ -10,7 +10,8 @@
     "noImplicitAny": true,
     "strict": true,
     "skipLibCheck": true,
-    "typeRoots": ["./node_modules/@types"]
+    "typeRoots": ["./node_modules/@types"],
+    "useUnknownInCatchVariables": false
```

Comment:
le type 'unkown' est arrivé dans les catch et ça casse notre gestion d'erreur. Cependant nos erreurs sont des erreurs Axios et donc elles sont bien typées. On peut faire un gros réfacto pas beau ou bien supprimer cette règle j'ai donc préféré supprimer cette règle. 
https://stackoverflow.com/questions/60151181/object-is-of-type-unknown-typescript-generics

Sur formation on a préféré typer toutes les erreurs en any mais je n'aime pas cette solution (et elle n'est pas possible sur outil)

Reply:
Sais-tu pour quels cas d'usage cette règle a été ajouté ?

Reply:
Je ne suis pas sure de comprendre ce qui casse notre gestion d'erreur, car dans la nouvelle version les types unknow remplacent les any mais dans notre gestion d'erreur on ne typait deja pas si?

Reply:
Chloé : c'est une règle qui oblige le typage des catch et selon leur doc : _This pattern ensures that error handling code becomes more comprehensive because you cannot guarantee that the object being thrown is a Error subclass ahead of time_
https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables
Et je ne trouve vraiment pas que ça améliore notre compréhension du code de gestion d'erreur.

Manon : les erreurs sont considérée comme any par défaut quand on ne type pas. c'est pour ça qu'il vaut mieux éviter les any car il n'y a pas vraiment de différence entre une variable non typée et une variable typée any.
Là les erreurs sont devenue des types 'unknown' obligatoirement. Et quand on veut accéder à un objet dans notre objet typé unknown, on a une erreur :
- e: any, e.response -> ok
- e sans type, e.response -> ok
- e: unknown, e.response -> ko

Reply:
Je crois que "comprehensive" c'est plutot exhaustif que comprehensif. J'ai l'impression qu'ils nous indiquent qu'en realite dans les catch on peut avoir autre chose que des erreurs (je ne sais pas bien dans quels cas d'ou ma question sur les cas d'usage, pour comprendre si ca peut nous arriver) et donc qu'il faut les gerer.

Reply:
Je ne vois pas trop non plus  quel cas pourrait renvoyer une erreur qui n'est pas une AxiosError. Nos gestions d'erreurs e.response viennent d'un appel à une route axios. Si la route axios appelée ne renvoie pas d'erreur axios, on aura une 500 car on essaye d'accéder à `e.response.message` avec `e.response` qui n'existe pas.
On pourrait faire une gestion d'erreur pour le cas où la réponse envoyée n'est pas de type AxiosError mais comme je le précise, c'est un gros refacto pour pas grand chose sachant que notre typescript est déjà loin d'être parfait, je pense qu'il y a d'autre réfacto plus important à mettre ne place avant

Reply:
Ok ! Peux-tu creer un ticket sur productboard pour qu'on garde une trace qu'il faudrait faire cette refacto ?